### DISCUSSION TITLE
Using PipeWire instead of Alsa

### DISCUSSION THREAD

**Discussion by haricane8133** [baresip/baresip#2891] [Q&A] [answered]
### Preface

_In the earlier Raspbian release (with v1.0.0 of BareSip), I had to tweak some Alsa configs to get things working - https://github.com/baresip/baresip/issues/808._

Now I am using the latest Raspbian release based on Debian BookWorm, which introduces PipeWire instead of PulseAudio+Alsa which existed in the earlier release.

I don't find official instructions on how to change the default audio engine that BareSip uses, from Alsa to PipeWire...

I just did a full manual build of the latest stable release of BareSip (3.8.1) on Raspberry Pi 4 running the latest Raspbian BookWorm 64-bit Lite OS (no GUI and based on ARM64).

Alsa as expected, did not work with the latest Raspbian.
```
dl: mod: /usr/local/lib/baresip/modules/alsa.so (/usr/local/lib/baresip/modules/alsa.so: cannot open shared object file: No such file or directory)
module alsa.so: No such file or directory [2]
```

Therefore I have to get the pipewire module working.

### What I tried

Within `$HOME/.baresip/config`, I tried commenting out module `alsa.so` within "**Audio Driver Modules**" and enabling module `pipewire.so`. I also changed the "`Audio`" settings to the following, as I saw vaguely mentioned in one of the PRs related to pipewire
```
audio_player            pipewire,default
audio_source            pipewire,default
audio_alert             pipewire,default
```

When I start Baresip after this change, I am shown a very similar error

```
dl: mod: /usr/local/lib/baresip/modules/pipewire.so (/usr/local/lib/baresip/modules/pipewire.so: cannot open shared object file: No such file or directory)
module pipewire.so: No such file or directory [2]
```

### Where I need help

Please let me know if I am doing the process of using another module, correctly.
Is there something I need to do to get pipewire working

**Comment by haricane8133** [comment]
is there something we need to pass as flags to the cmake script?
`cmake -B build -DMODULES="pipewire;..."`

Would the following command build all the modules available?
`cmake -B build -DCMAKE_BUILD_TYPE=Release`

**Reply by sreimers** [reply, marked as answer]
Modules by default only build if all dev dependencies are detected, for alsa this needs at least `libasound2-dev` on debian. Can you provide the output of  `cmake -B build -DCMAKE_BUILD_TYPE=Release`

For pipewire I think this is: `libpipewire-0.3-dev`

**Reply by haricane8133** [reply]
@sreimers, Thanks!
